zebra-chain/src/parameters/network.rs

```diff
@@ -51,52 +53,105 @@ mod tests;
+    /// The production mainnet.
+    #[default]
+    Mainnet,
+
+    /// A test network.
+    Testnet,
+}
+
+impl From<Network> for NetworkKind {
+    fn from(network: Network) -> Self {
+        network.kind()
+    }
+}
+
+/// An enum describing the possible network choices.
+#[derive(Clone, Debug, Default, Eq, PartialEq, Hash, Serialize)]
+#[cfg_attr(any(test, feature = "proptest-impl"), derive(Arbitrary))]
+#[serde(into = "NetworkKind")]
 pub enum Network {
     /// The production mainnet.
     #[default]
     Mainnet,
 
     /// The oldest public test network.
-    Testnet,
+    Testnet(Arc<NetworkParameters>),
 }
 
-use zcash_primitives::consensus::{Network as ZcashPrimitivesNetwork, Parameters as _};
-impl Network {
+impl NetworkKind {
     /// Returns the human-readable prefix for Base58Check-encoded transparent
     /// pay-to-public-key-hash payment addresses for the network.
-    pub fn b58_pubkey_address_prefix(&self) -> [u8; 2] {
+    pub fn b58_pubkey_address_prefix(self) -> [u8; 2] {
         <ZcashPrimitivesNetwork>::from(self).b58_pubkey_address_prefix()
     }
 
     /// Returns the human-readable prefix for Base58Check-encoded transparent pay-to-script-hash
     /// payment addresses for the network.
-    pub fn b58_script_address_prefix(&self) -> [u8; 2] {
+    pub fn b58_script_address_prefix(self) -> [u8; 2] {
         <ZcashPrimitivesNetwork>::from(self).b58_script_address_prefix()
     }
-    /// Returns true if the maximum block time rule is active for `network` and `height`.
-    ///
-    /// Always returns true if `network` is the Mainnet.
-    /// If `network` is the Testnet, the `height` should be at least
-    /// TESTNET_MAX_TIME_START_HEIGHT to return true.
-    /// Returns false otherwise.
-    ///
-    /// Part of the consensus rules at <https://zips.z.cash/protocol/protocol.pdf#blockheader>
-    pub fn is_max_block_time_enforced(&self, height: block::Height) -> bool {
+
+    /// Return the network name as defined in
+    /// [BIP70](https://github.com/bitcoin/bips/blob/master/bip-0070.mediawiki#paymentdetailspaymentrequest)
+    pub fn bip70_network_name(&self) -> String {
         match self {
-            Network::Mainnet => true,
-            Network::Testnet => height >= super::TESTNET_MAX_TIME_START_HEIGHT,
+            Self::Mainnet => "main".to_string(),
+            Self::Testnet => "test".to_string(),
+        }
+    }
+
+    /// Converts a [`zcash_address::Network`] to a [`NetworkKind`].
+    pub fn from_zcash_address(network: zcash_address::Network) -> Self {
+        match network {
+            zcash_address::Network::Main => NetworkKind::Mainnet,
+            zcash_address::Network::Test | zcash_address::Network::Regtest => NetworkKind::Testnet,
+        }
+    }
+}
+
+impl From<NetworkKind> for &'static str {
+    fn from(network: NetworkKind) -> &'static str {
+        match network {
+            NetworkKind::Mainnet => "MainnetKind",
+            NetworkKind::Testnet => "TestnetKind",
```

Comment:
Consider adding a correctness comment that this shouldn't be "Testnet" since it's too close to the lowercase form which could be parsed as the default public testnet.

Consider adding a test verifying that this variant can't be parsed or converted `.into()` the default testnet.